File: B3WM/B3WM.Client/Model/BubbleStorageItem.cs
using Magic.IndexedDb;
using Magic.IndexedDb.SchemaAnnotations;
using static B3WM.Client.Model.BubbleStorageItem;
using static B3WM.Shared.Entity.Ticks2;

namespace B3WM.Client.Model
{
    /// <summary>
    /// Bubbles para IndexedDB. KeyPath "id" (case-sensitive). Sem timeframe.
    /// </summary>
    public class BubbleStorageItem : MagicTableTool<BubbleStorageItem>, IMagicTable<DbSets>
    {
        public static readonly IndexedDbSet DataBase = IndexDbContext.DataBase;

        public List<IMagicCompoundIndex> GetCompoundIndexes() =>
        new List<IMagicCompoundIndex>() {
            CreateCompoundIndex(x=> Date,
                X=> Agent,
                x=> ActionType,
                X=> Price,
                x => Symbol)
        };

        public IMagicCompoundKey GetKeys() =>
            CreateCompoundKey(
                x=> Date,
                X=> Agent,
                x=> ActionType,
                X=> Price,
                x => Symbol
                );
        //CreatePrimaryKey(x => x.Id, true); // Auto-incrementing primary key

        public string GetTableName() => "Bubbles";
        public IndexedDbSet GetDefaultDatabase() => IndexDbContext.DataBase;

        public DbSets Databases { get; } = new();
        public sealed class DbSets
        {
            public readonly IndexedDbSet BubbleStorageItem = IndexDbContext.DataBase;
        }

        [MagicName("Price")]
        public double Price { get; set; }

        [MagicName("Agent")]
        public int Agent { get; set; }

        [MagicName("Amount")]
        public decimal Amount { get; set; }

        [MagicName("Date")]
        public DateTime Date { get; set; }

        [MagicName("actionType")]
        public ActionType ActionType { get; set; }

        [MagicName("Symbol")]
        public string Symbol { get; set; }
    }
}

File: B3WM.Flutter/lib/models/bubble_storage_item.dart
import 'ticks2.dart';

class BubbleStorageItem {
  final double price;
  final int agent;
  final double amount;
  final DateTime date;
  final ActionType actionType;
  final String symbol;

  BubbleStorageItem({
    required this.price,
    required this.agent,
    required this.amount,
    required this.date,
    required this.actionType,
    required this.symbol,
  });

  factory BubbleStorageItem.fromJson(Map<String, dynamic> json) {
    return BubbleStorageItem(
      price: (json['price'] as num).toDouble(),
      agent: json['agent'] as int,
      amount: (json['amount'] as num).toDouble(),
      date: DateTime.parse(json['date'] as String),
      actionType: ActionType.fromValue(json['actionType']),
      symbol: json['symbol'] as String,
    );
  }

  Map<String, dynamic> toJson() => {
        'price': price,
        'agent': agent,
        'amount': amount,
        'date': date.toIso8601String(),
        'actionType': actionType.value,
        'symbol': symbol,
      };
}
